gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/CopySource.java

```diff
@@ -366,6 +372,9 @@ public Void call() {
               workUnit.setProp(ConfigurationKeys.COPY_EXPECTED_SCHEMA, ((ConfigBasedDataset) this.copyableDataset).getExpectedSchema());
             }
           }
+          if ((this.copyableDataset instanceof HiveDataset) && (state.getPropAsBoolean(ConfigurationKeys.DATASET_STAGING_DIR,false))) {
```

Comment:
Sorry If we have discussed this before, but what's the reason that you couldn't set `DATASET_STAGING_DIR_PATH` in the Hive-specific constructs like `HiveDataset`?